proposals/p2240.md

```diff
@@ -656,90 +528,95 @@ starting with `Q` (corresponding to the cases where `x` and/or `z` are empty).
-
-There must be at least N+M non-variadic argument expressions, because otherwise
-if all variadic argument expressions are empty, there will not be enough
-arguments to match all the non-variadic parameters. We will refer to the Nth
-non-variadic argument expression and the argument expressions before it as
-"leading argument expressions". Similarly, we will refer to the Mth-from-last
-non-variadic argument expression and the argument expressions after it as
-"trailing argument expressions", and any remaining argument expressions as
-"central argument expressions". A "leading argument" is an argument that was
-produced by rewriting a leading argument expression, and likewise for "central
-argument" and "trailing argument".
-
-By construction, there will always be at least N leading arguments, because
-there are N non-variadic leading argument expressions. Likewise, there will
-always be at least M trailing arguments. As a result, a leading parameter can
-only match a leading argument, and so it can only match a leading argument
-expression, and likewise for trailing parameters. Consequently, a leading
-argument expression cannot match a trailing parameter, a trailing argument
-expression cannot match a leading parameter, and a central argument expression
-can only match the variadic parameter.
-
-Consider the i'th non-variadic leading argument expression E. If all the
-variadic argument expressions before it are empty, E will match the i'th leading
-parameter, so E cannot match any earlier parameter. If there are any earlier
-variadic argument expressions, E can be made to match any later leading
-parameter or the variadic parameter, by making one of those earlier variadic
-arguments large enough, but as observed above, E cannot match a trailing
-parameter. If there are no earlier variadic argument expressions, E cannot be
-made to match any later parameter, so it can only match the i'th leading
-parameter.
-
-Next, consider a variadic leading argument expression E that comes before the
-i'th non-variadic leading argument expression, but not before any earlier
-non-variadic argument expression. If E's rewritten size is sufficiently large,
-and all earlier variadic argument expressions are empty, it will simultaneously
-match the i'th leading parameter, all leading parameters after it, and the
-variadic parameter, but as before, it cannot match a trailing parameter.
-
-The same reasoning can be applied to trailing argument expressions, but with
-"before" and "after" reversed. And as noted earlier, central argument
-expressions can only match the variadic parameter. In summary, we can identify
-the possible matches for an argument expression E as follows:
-
--   If E is leading, let i be one more than the number of earlier non-variadic
-    argument expressions:
-    -   If E is non-variadic, and there are no earlier variadic argument
-        expressions, then E can only match the i'th leading parameter.
+##### Identifying potential matchings
```

Comment:
Might be good to use some different formatting for the variables N, M, i, etc., like _italics_ or `code`.